### PR TITLE
feat(presto-on-hive): allow v1 fieldpaths in the presto-on-hive source

### PR DESCRIPTION
**Pull request by gabe-lyons**
## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

**Review by asikowitz — COMMENTED** [review]
As mentioned in slack, would like to use `_get_simple_field_path_from_v2_field_path` in `DatasetUrn` or merge functionality. Also, think @hsheth2 would probably have a better idea here

File: metadata-ingestion/src/datahub/ingestion/source/sql/sql_common.py [hunk 1]
```diff
@@ -287,7 +289,12 @@ def get_schema_metadata(
     pk_constraints: Optional[dict] = None,
     foreign_keys: Optional[List[ForeignKeyConstraint]] = None,
     canonical_schema: Optional[List[SchemaField]] = None,
+    simplify_nested_field_paths: bool = False,
 ) -> SchemaMetadata:
+    if simplify_nested_field_paths:
+        if not schema_requires_v2(canonical_schema):
+            canonical_schema = downgrade_schema_from_v2(canonical_schema)
```
[the remark this answers is not in the document]
Comment (gabe-lyons) [reply metadata-ingestion/src/datahub/ingestion/source/sql/sql_common.py]: resolved!

File: metadata-ingestion/src/datahub/ingestion/source/sql/sql_utils.py [hunk 2]
```diff
@@ -223,3 +229,43 @@ def gen_lineage(
 
         for wu in lineage_workunits:
             yield wu
+
+
+# downgrade a schema field path
+def downgrade_v2_field_path(field_path: str) -> str:
```
Comment (hsheth2) [inline metadata-ingestion/src/datahub/ingestion/source/sql/sql_utils.py:235]: yep it'd be good to replace this with `DatasetUrn._get_simple_field_path_from_v2_field_path`

**Review by hsheth2 — COMMENTED** [review]
lgtm once we switch to using `_get_simple_field_path_from_v2_field_path`

**Review by hsheth2 — APPROVED** [review]

**Pull request merged**